File: app/src/main/java/com/chad/baserecyclerviewadapterhelper/base/BaseDataBindingAdapter.java
package com.chad.baserecyclerviewadapterhelper.base;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.LayoutRes;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;

import com.chad.library.adapter.base.BaseQuickAdapter;

import org.jetbrains.annotations.NotNull;

import java.util.List;


/**
 * Created by tysheng
 * Date: 2017/5/11 14:39.
 * Email: tyshengsx@gmail.com
 */

public abstract class BaseDataBindingAdapter<T, B extends ViewDataBinding> extends BaseQuickAdapter<T, BaseBindingViewHolder<B>> {


    public BaseDataBindingAdapter(@LayoutRes int layoutResId, @Nullable List<T> data) {
        super(layoutResId, data);
    }

    public BaseDataBindingAdapter(@Nullable List<T> data) {
        super(0, data);
    }

    public BaseDataBindingAdapter(@LayoutRes int layoutResId) {
        super(layoutResId);
    }

//    @NonNull
//    @Override
//    protected BaseBindingViewHolder<B> createBaseViewHolder(@NonNull View view) {
//        return new BaseBindingViewHolder<>(view);
//    }

    @NotNull
    @Override
    protected BaseBindingViewHolder<B> createBaseViewHolder(@NotNull ViewGroup parent, int layoutResId) {
        B b = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()), layoutResId, parent, false);
        View view;
        if (b == null) {
            view = LayoutInflater.from(parent.getContext()).inflate(layoutResId, parent);
        } else {
            view = b.getRoot();
        }
        BaseBindingViewHolder<B> holder = new BaseBindingViewHolder<>(view);
        holder.setBinding(b);
        return holder;
    }


    @Override
    protected void convert(@NonNull BaseBindingViewHolder<B> helper, T item) {
        convert(helper.getBinding(), item);
        helper.getBinding().executePendingBindings();
    }

    protected abstract void convert(B b, T item);
}

File: kotlinapp/src/main/java/com/allen/kotlinapp/base/BaseDataBindingAdapter.kt
package com.allen.kotlinapp.base

import android.databinding.DataBindingUtil
import android.databinding.ViewDataBinding
import android.support.annotation.LayoutRes
import android.view.View
import android.view.ViewGroup
import com.chad.library.adapter.base.BaseQuickAdapter

/**
 * 文 件 名: BaseDataBindingAdapter
 * 创 建 人: Allen
 * 创建日期: 2017/6/13 15:05
 * 修改时间：
 * 修改备注：
 */
abstract class BaseDataBindingAdapter<T, Binding : ViewDataBinding> : BaseQuickAdapter<T, BaseBindingViewHolder<Binding>> {


    constructor(@LayoutRes layoutResId: Int, data: List<T>?) : super(layoutResId, data) {}

    constructor(data: List<T>?) : super(data) {}

    constructor(@LayoutRes layoutResId: Int) : super(layoutResId) {}

    override fun createBaseViewHolder(view: View): BaseBindingViewHolder<Binding> {
        return BaseBindingViewHolder(view)
    }

    override fun createBaseViewHolder(parent: ViewGroup, layoutResId: Int): BaseBindingViewHolder<Binding> {
        val binding = DataBindingUtil.inflate<Binding>(mLayoutInflater, layoutResId, parent, false)
        val view: View
        if (binding == null) {
            view = getItemView(layoutResId, parent)
        } else {
            view = binding.root
        }
        val holder = BaseBindingViewHolder<Binding>(view)
        holder.binding = (binding)
        return holder
    }

    override fun convert(helper: BaseBindingViewHolder<Binding>, item: T) {
        convert(helper.binding!!, item)
        helper.binding!!.executePendingBindings()
    }

    protected abstract fun convert(binding: Binding, item: T)
}
